app/Http/Controllers/LocationController.php

```diff
@@ -2,9 +2,126 @@
 
 namespace App\Http\Controllers;
 
+use App\Models\Address;
+use App\Models\Location;
 use Illuminate\Http\Request;
 
 class LocationController extends Controller
 {
-    //
+
+    //Public
+    public function publicIndex(){
+        return view('locations.public.index',['locations'=>Location::all()]);
+    }
+
+    public function publicShow(int $id){
+        return view('locations.public.show',['location'=>Location::findOrFail($id)]);
+    }
+
+    //Admin
+    public function create()
+    {
+        return view('locations.admin.create');
+    }
+
+    public function store(Request $request){
+        $request->validate([
+            'short_description' => 'required|string|min:2',
+            'long_description'=>'string',
+            'name' => 'required|string|min:2',
+            'cover_img_path'=>'nullable|image|mimes:jpeg,png|max:2048',
+            'street_number' =>'string|required',
+            'street_name' => 'string|required',
+            'zip_code'=> 'string|required',
+            'city'=>'string|required',
+            'country'=>'string|required'
+        ]);
+        $address = Address::create([
```

Comment:
What if address already exists in database?
Like if a single address have multiple danse sale

Hint: Du kan bruge "firstOrCreate()"

Reply:
Er rettet